evm/src/cpu/stack.rs

```diff
@@ -0,0 +1,387 @@
+use itertools::izip;
+use plonky2::field::extension::Extendable;
+use plonky2::field::packed::PackedField;
+use plonky2::field::types::Field;
+use plonky2::hash::hash_types::RichField;
+use plonky2::iop::ext_target::ExtensionTarget;
+
+use crate::constraint_consumer::{ConstraintConsumer, RecursiveConstraintConsumer};
+use crate::cpu::columns::ops::OpsColumnsView;
+use crate::cpu::columns::CpuColumnsView;
+use crate::cpu::membus::NUM_GP_CHANNELS;
+use crate::memory::segments::Segment;
+
+#[derive(Clone, Copy)]
+struct StackBehavior {
+    num_pops: usize,
+    pushes: bool,
+    disable_other_channels: bool,
```

Comment:
This would be false for things like `MLOAD` right?

To be extra safe we could do something like `extra_channels_used: usize`. Otherwise it seems some instructions like MLOAD would need to manually disable the last couple channels?

Reply:
> This would be false for things like `MLOAD` right?

Exactly!

> To be extra safe we could do something like `extra_channels_used: usize`.

I agree, but I’d like to call it out of the scope of this PR. This is the kind of implementation detail that I wouldn’t want to see in `stack.rs`, but yes we should do this somewhere